engine/dlib/src/dlib/connection_pool.cpp

```diff
@@ -407,7 +407,7 @@ namespace dmConnectionPool
         // try connecting to the host using ipv4 first
         uint64_t dial_started = dmTime::GetTime();
         Result r = DoDial(pool, host, port, ssl, timeout, cancelflag, connection, sock_res, 1, 0);
-        if (r == RESULT_OK || r == RESULT_SHUT_DOWN || r == RESULT_OUT_OF_RESOURCES)
```

Comment:
One more change here.
He we try to connect ipv4 and if it's failed we do repeat connection using ipv6.
But this logic doesn't take into account `RESULT_HANDSHAKE_FAILED` error and repeat request using ipv6 in this case. I fixed that to avoid double request.